docs/src/_guide/configuration.md

```diff
@@ -68,14 +68,72 @@ export <PLUGIN_NAME>_<SETTING_NAME>=<value>
 export TAP_GITLAB_API_URL=https://gitlab.example.com
 ```
 
-Plugins can also specify alternative variables (aliases) for their settings, to match existing usage or variables expected by plugin executables. You can use [`meltano config <plugin> list`](/reference/command-line-interface#config) to list all available settings for a plugin along with their variables, in order of precedence.
+Plugins can also specify alternative variables ([aliases](#aliases) for their settings, to match existing usage or variables expected by plugin executables. You can use [`meltano config <plugin> list`](/reference/command-line-interface#config) to list all available settings for a plugin along with their variables, in order of precedence.
 
 Since environment variable values are always strings, Meltano will cast values to the appropriate type before passing them on to the plugin.
 
 To verify that any environment variables you've set will be picked up by Meltano as you intended, you can test them with [`meltano config <plugin>`](/reference/command-line-interface#config) before running [`meltano elt`](/reference/command-line-interface#elt) or [`meltano invoke`](/reference/command-line-interface#invoke).
 
 To learn how to use environment variables to specify pipeline-specific configuration, refer to the [Data Integration (EL) guide](/guide/integration#pipeline-specific-configuration).
 
+#### aliases
+
+Aliases allow for configuration values to be set via one of multiple keys.
+Environment variable aliases are listed next to the canonical names for the variable in the output of the [`meltano config <plugin> list`](/reference/command-line-interface#config) command.
+They can be defined via the `aliases` key in a custom plugin's `settings` configuration.
+For example, the following defines a `my_custom_username` setting with aliases `custom_tap_username` and `username`:
+
+```yaml
+# meltano.yml
+...
+plugins:
+  extractors:
+  - name: my-custom-tap
+    namespace: my_custom_tap
+    pip_url: git+https://github.com/my-organization/my-custom-tap.git
+    executable: my-custom-tap
+    capabilities:
+    - discover
+    - catalog
+    settings:
+    - name: password
+      kind: password
+    - name: my_custom_tap_username
+      aliases: [custom_tap_username, username]
+```
+
+Within a given configuration layer, a setting can be set via only a single name, whether that name is its canonical name or one of its aliases.
+So given the custom extractor defined above, the `my_custom_tap_username` setting could be set via the `MY_CUSTOM_TAP_MY_CUSTOM_TAP_USERNAME` environment variable or either the `MY_CUSTOM_TAP_CUSTOM_TAP_USERNAME` or `MY_CUSTOM_TAP_USERNAME` variables.
+But if more than one of these variables is set in the terminal environment, then an exception will be raised--even if all the relevant environment variables have the same value.
+
```

Comment:
Can we add a note in here about being able to reference the alias when using `meltano config`? Basically making it clear that 

`meltano config <plugin> set <name>`

and

`meltano config <plugin> set <alias>`

are both ok to do.